src/k8s-engine/wsl.ts

```diff
@@ -412,16 +424,7 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
       .map(result => result.groups?.name)
       .filter(defined);
 
-    if (runningOnly) {
-      const runningOut = await this.execWSL({ capture: true }, '--list', '--quiet', '--running');
-      const running = runningOut.split(/[\r\n]+/).map(x => x.trim()).filter(x => x);
-
-      result = result.filter(x => running.includes(x));
-    }
-
-    console.debug(`${ runningOnly ? 'Running' : 'All' } distros: ${ result }`);
-
-    return result;
+    return result.filter(x => distros.includes(x));
```

Comment:
What about `return !runningOnly ? result : result.filter(x => distros.includes(x))` ?
More code, but no need to intersect the lists if we want all of them.

Reply:
Makes sense; my tree is busy, but I think I'll do `return result.filter(x => runningOnly || distros.includes(x));` (relying on short circuiting to bypass the extra evaluation).

Reply:
I don't know; this logic is kind of hard to understand, and you are optimizing a filter of less than 20 items.

Given that we don't have tests for the logic, I would prefer the simpler version. We already spend more time "talking" about this than this optimization will ever save us.

Reply:
Alright, given the short expected list, I'll keep things as-is; we can revisit if people report having a bazillion WSL distros.